tests/e2e/volume_provisioning_with_level5_topology.go

```diff
@@ -0,0 +1,1237 @@
+		ginkgo.By(fmt.Sprintf("Verify volume: %s is attached to the node: %s", pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName))
+		vmUUID = getNodeUUID(client, pod.Spec.NodeName)
+		isDiskAttached, err := e2eVSphere.isVolumeAttachedToVM(client, volHandle, vmUUID)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(isDiskAttached).To(gomega.BeTrue(), "Volume is not attached to the node")
+		defer func() {
+			ginkgo.By(fmt.Sprintf("Deleting the pod %s in namespace %s", pod.Name, namespace))
+			err = fpod.DeletePodWithWait(client, pod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By("Verify volume is detached from the node")
+			isDiskDetached, err := e2eVSphere.waitForVolumeDetachedFromNode(client, pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(isDiskDetached).To(gomega.BeTrue(),
+				fmt.Sprintf("Volume %q is not detached from the node %q", pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName))
+		}()
+
+		/* Verify PV node affinity and that the PODS are running on appropriate node as
+		specified in the allowed topologies of SC */
+		ginkgo.By("Verify PV node affinity and that the PODS are running on " +
+			"appropriate node as specified in the allowed topologies of SC")
+		verifyPVnodeAffinityAndPODnodedetailsFoStandalonePodLevel5(ctx, client, pod, namespace, allowedTopologies)
+	})
+
+	/*
+		TESTCASE-6
+		Steps:
+		1. Create FCD on the datastore present in datacenter > cluster2
+		2. Create SC which points to region1 > zone1 > building1 > level1 > rack > rack2.
+		3. Create PV using the above created FCD, SC and 'nodeSelectorTerms' of SC.
+		4. Create PVC using above created SC and PV.
+		5. Wait for PV, PVC to be in bound.
+		6. Describe  PV and verify node affinity details. It should display all 5 levels.
+		7. Create POD using the above PVC.
+		8. POD should come up on the node which is present in the same zone as mentioned in the storage class.
+		9. Delete POD, PVC, PV and SC.
+	*/
+	ginkgo.It("Verify static volume provisioning with FCD and storage class with allowed topologies", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		/* Get allowed topologies for Storage Class
+		region1 > zone1 > building1 > level1 > rack > rack2 */
+		allowedTopologyForSC := getTopologySelector(topologyAffinityDetails, topologyCategories,
+			topologyLength, leafNode, leafNodeTag1)
+
+		// Create SC with Immediate BindingMode and allowed topology set to 5 levels
+		storageclass, err := createStorageClass(client, nil, allowedTopologyForSC, "", "", false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Creating FCD Disk")
+		fcdID, err := e2eVSphere.createFCD(ctx, "BasicStaticFCD", diskSizeInMb, nonsharedDatastore.Reference())
```

Comment:
Why non-shared datastore?

Reply:
same here.
non-shared here means cluster2 ds and volume provisioning will happen on cluster2 ds.